lib/ts/recipe/session/index.ts

```diff
@@ -52,6 +52,19 @@ export default class SessionWrapper {
         });
     }
 
+    static getSession(req: any, res: any): Promise<SessionContainer>;
+    static getSession(
+        req: any,
+        res: any,
+        options?: VerifySessionOptions & { sessionRequired?: true },
+        userContext?: any
+    ): Promise<SessionContainer>;
+    static getSession(
+        req: any,
+        res: any,
+        options?: VerifySessionOptions & { sessionRequired: false },
+        userContext?: any
+    ): Promise<SessionContainer | undefined>;
```

Comment:
Very interesting. :thinking: When I inspect the `Session` from `const Session = require("supertokens-node/recipe/session");` by performing a `ctrl + click`, VS Code takes me to the file that I edited in this PR.

Should I _also_ add the same type definitions to these other files that you mentioned? Is there a way to re-use these type definitions so that we only have to write them once?

I'm curious to know why VS Code takes me to a file different from the one you gave. :thinking:

Reply:
> Should I also add the same type definitions to these other files that you mentioned? Is there a way to re-use these type definitions so that we only have to write them once?

Hmm. On second thought, this is the only place that needs the change, since the user interacts only with this function directly. So I guess this is good to go.

> I'm curious to know why VS Code takes me to a file different from the one you gave.

VSCode does the right thing. When you import from `/recipe/session`, you import from the index.ts file you made the change to.

Reply:
Okay cool. I _think_ I've addressed all of your comments. Is there anything else you need from me in this PR?